src/v0/destinations/iterable/util.js

```diff
@@ -744,6 +748,93 @@ const filterEventsAndPrepareBatchRequests = (transformedEvents) => {
+const isValueInResponseArray = (path, value, response) => {
+  const responseValueArray = getValueFromResponse(path, response);
+  return Array.isArray(responseValueArray) && responseValueArray.includes(value);
+};
+
+/**
+ * Determines if an event should be aborted based on the response from a destination
+ * and extracts an error message if applicable.
+ * ref:
+ * 1) https://api.iterable.com/api/docs#users_updateEmail
+ * 2) https://api.iterable.com/api/docs#events_track
+ * 3) https://api.iterable.com/api/docs#users_bulkUpdateUser
+ * 4) https://api.iterable.com/api/docs#events_trackBulk
+ * 5) https://api.iterable.com/api/docs#catalogs_bulkUpdateCatalogItems
+ * 6) https://api.iterable.com/api/docs#users_registerDeviceToken
+ * 7) https://api.iterable.com/api/docs#users_registerBrowserToken
+ * 8) https://api.iterable.com/api/docs#commerce_trackPurchase
+ * 9) https://api.iterable.com/api/docs#commerce_updateCart
+ *
+ * @param {Object} event - The event object containing various event properties.
+ * @param {Object} destinationResponse - The response object from the destination.
+ * @returns {Object} An object containing a boolean `isAbortable` indicating if the event
+ * should be aborted, and an `errorMsg` string with the error message if applicable.
+ */
+const checkIfEventIsAbortableAndExtractErrorMessage = (event, destinationResponse) => {
+  const { failCount } = destinationResponse.response;
+
+  if (failCount === 0) {
+    return { isAbortable: false, errorMsg: '' };
+  }
+
+  const eventValues = {
+    email: event.email,
+    userId: event.userId,
+    eventName: event.eventName,
+  };
+
+  const matchingPath =
+    ITERABLE_RESPONSE_USER_ID_PATHS.find((userIdPath) =>
+      isValueInResponseArray(userIdPath, eventValues.userId, destinationResponse.response),
+    ) ||
+    ITERABLE_RESPONSE_EMAIL_PATHS.find((emailPath) =>
+      isValueInResponseArray(emailPath, eventValues.email, destinationResponse.response),
+    ) ||
+    (isValueInResponseArray(
+      'disallowedEventNames',
+      eventValues.eventName,
+      destinationResponse.response,
+    )
+      ? 'disallowedEventNames'
+      : null);
+
+  if (matchingPath) {
+    const responseValueArray = getValueFromResponse(matchingPath, destinationResponse.response);
+    const matchingValue = responseValueArray.find((value) => {
+      if (ITERABLE_RESPONSE_EMAIL_PATHS.some((emailPath) => matchingPath.includes(emailPath))) {
+        return value === eventValues.email;
+      }
+      if (ITERABLE_RESPONSE_USER_ID_PATHS.some((userIdPath) => matchingPath.includes(userIdPath))) {
+        return value === eventValues.userId;
+      }
+      if (matchingPath === 'disallowedEventNames') {
+        return value === eventValues.eventName;
+      }
+      return false;
+    });
+
+    return {
+      isAbortable: true,
+      errorMsg: `Request failed for value "${matchingValue}" because it is "${matchingPath}".`,
+    };
+  }
+
+  return { isAbortable: false, errorMsg: '' };
+};
```

Comment:
```suggestion
/**
 * Checks if a value is present in a response array based on a given path.
 * @param {Object} response - The response object to search within.
 * @param {string} path - The path to the response array.
 * @param {any} value - The value to check for in the array.
 * @returns {boolean} - True if the value is in the array, otherwise false.
 */
const isValueInResponseArray = (response, path, value) => {
  const respArr = get(response, path);
  return Array.isArray(respArr) && respArr.includes(value);
};

/**
 * Determines if an event should be aborted based on the response from a destination
 * and extracts an error message if applicable.
 * ref:
 * 1) https://api.iterable.com/api/docs#users_updateEmail
 * 2) https://api.iterable.com/api/docs#events_track
 * 3) https://api.iterable.com/api/docs#users_bulkUpdateUser
 * 4) https://api.iterable.com/api/docs#events_trackBulk
 * 5) https://api.iterable.com/api/docs#catalogs_bulkUpdateCatalogItems
 * 6) https://api.iterable.com/api/docs#users_registerDeviceToken
 * 7) https://api.iterable.com/api/docs#users_registerBrowserToken
 * 8) https://api.iterable.com/api/docs#commerce_trackPurchase
 * 9) https://api.iterable.com/api/docs#commerce_updateCart
 *
 * @param {Object} event - The event object containing various event properties.
 * @param {Object} destinationResponse - The response object from the destination.
 * @returns {Object} An object containing a boolean `isAbortable` indicating if the event
 * should be aborted, and an `errorMsg` string with the error message if applicable.
 */
const checkIfEventIsAbortableAndExtractErrorMessage = (event, destinationResponse) => {
  const { failCount } = destinationResponse.response;

  if (failCount === 0) {
    return { isAbortable: false, errorMsg: '' };
  }

  const eventValues = {
    email: event.email,
    userId: event.userId,
    eventName: event.eventName,
  };

  let errorMsg = '';
  const userIdMatchPath = ITERABLE_RESPONSE_USER_ID_PATHS.filter((userIdPath) =>
    isValueInResponseArray(destinationResponse.response, userIdPath, eventValues.userId),
  );
  errorMsg += `userId error:"${eventValues.userId}" in "${userIdMatchPath}".`;

  const emailMatchPath = ITERABLE_RESPONSE_EMAIL_PATHS.filter((emailPath) =>
    isValueInResponseArray(destinationResponse.response, emailPath, eventValues.email),
  );
  errorMsg += `email error:"${eventValues.email}" in "${emailMatchPath}".`;

  const eventNameMatchPath = ['disallowedEventNames'].filter((eventNamePath) =>
    isValueInResponseArray(destinationResponse.response, eventNamePath, eventValues.eventName),
  );
  errorMsg += `eventName error:"${eventValues.eventName}" in "${eventNameMatchPath}".`;

  if (errorMsg) {
    return {
      isAbortable: true,
      errorMsg,
    };
  }

  return { isAbortable: false, errorMsg: '' };
};
```

The intention is to collect all the error messages and display the same to the user.
Feel free to update error message, variable names, function names etc,.

cc: @utsabc